app/src/main/java/com/trianglz/weatherapp/domain/models/weather/Weather.kt

```diff
@@ -0,0 +1,37 @@
+package com.trianglz.weatherapp.domain.models.weather
+
+import com.trianglz.weatherapp.R
+import kotlin.random.Random
+
+data class Weather(
+    val countryCode: String,
+    val cityName: String,
+    val currentTemperature: Int,
+    val feelsLike: Int,
+    val lowTemperature: Int,
+    val highTemperature: Int,
+    val humidityPercentage: Int,
+    val cloudPercentage: Int,
+    val windSpeed: Double,
+    val windDegree: Int,
+    val sunrise: Long,
+    val sunset: Long,
+) {
+    fun getWeatherDescription(): String = when {
+        currentTemperature >= 30.0 -> "Hot."
+        currentTemperature >= 20.0 -> "Warm."
+        currentTemperature >= 10.0 -> "Cool."
+        currentTemperature >= 0.0 -> "Freezing!"
+        else -> "Stay indoors."
+    }
+
+    fun getWeatherIcon(): Int {
```

Comment:
Weather icon is UI related property. It is preferred to have a separate model for the UI related aspects.